### PR TITLE
HG 844 | add TOCT03, TOCT04, TOCT05

### PR DESCRIPTION
**Pull request by tomasznapieralski**
Algorithm TOCT03:
- navigate to article with TOC and with `#id` in url
- check that page was scrolled to proper header

Algorithm TOCT04:
- navigate to article with TOC
- click on link with redirection to current article with `#id` in link
- check that page was scrolled to proper header

Algorithm TOCT05:
- navigate to article without TOC
- click on link with redirection to article with TOC with `#id` in link
- check that page was scrolled to proper header

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TOCTests.java [hunk 1]
```diff
@@ -26,7 +28,7 @@ public void prepareTest() {
   private final static int H2_PADDING_TOP = 40;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TOCTests.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues:
* MAJOR 2 major
* MINOR 8 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Define a constant instead of duplicating this literal " was opened" 4 times.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L31) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Remove this unused import 'org.openqa.selenium.WebDriverException'.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L13) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L129) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L72) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Remove this unused import 'org.openqa.selenium.support.FindBy'.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L15) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Reorder the modifiers to comply with the Java Language Specification.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java#L22) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)
* MINOR [Define a constant instead of duplicating this literal "does not work" 4 times.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/testcases/mercurytests/TOCTests.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "works" 4 times.](https://github.com/Wikia/selenium-tests/blob/e5bba1cbf4cffb99dff25129ad50b73cee995f84/src/test/java/com/wikia/webdriver/testcases/mercurytests/TOCTests.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

**Comment by tomasznapieralski** [comment]
All tests passed on Jenkins
http://qa-s2.wikia-prod:8080/view/Mercury/job/Mercury-TESTS/98/artifact/logs/log.html

**Comment by lizlux** [comment]
LGTM thanks!

**Pull request merged**